From 4b02d02d26ed8586e90fc231720a406916f8cf0e Mon Sep 17 00:00:00 2001
From: Santiago Rossi <santiago_rho_psi@hotmail.com>
Date: Tue, 1 May 2012 01:52:02 -0300
Subject: [PATCH 21/40] =?UTF-8?q?4.3.2=20Torneo=20Argenta:=20El=20Desaf=C3=AD?=
 =?UTF-8?q?o=20del=20Aspirante?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixed quest "The aspirant's challenge" from Argent Tournamet quest chain
---
 ...012_05_01_100_world_the_aspirants_challenge.sql |   60 +++++++++
 src/server/scripts/Northrend/icecrown.cpp          |  131 --------------------
 2 files changed, 60 insertions(+), 131 deletions(-)
 create mode 100644 sql/updates/world/reanemu/2012_05_01_100_world_the_aspirants_challenge.sql

diff --git a/sql/updates/world/reanemu/2012_05_01_100_world_the_aspirants_challenge.sql b/sql/updates/world/reanemu/2012_05_01_100_world_the_aspirants_challenge.sql
new file mode 100644
index 0000000..3b0a1b1
--- /dev/null
+++ b/sql/updates/world/reanemu/2012_05_01_100_world_the_aspirants_challenge.sql
@@ -0,0 +1,60 @@
+-- Argent Valiant SAI
+SET @VALIANT := 33448;
+UPDATE `creature_template` SET `faction_A`=14, `faction_H`=14, `AIName`='SmartAI', `equipment_id`=1927, `ScriptName`='' WHERE `entry`=@VALIANT;
+UPDATE `creature_template_addon` SET `auras`= '63500 65190' WHERE `entry`=@VALIANT;
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@VALIANT AND `source_type`=0;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@VALIANT*100 AND `source_type`=9;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@VALIANT,0,0,0,4,0,100,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Argent Valiant - Aggro - Say Text'),
+(@VALIANT,0,1,0,0,0,100,0,10000,10000,10000,10000,11,65147,0,0,0,0,0,2,0,0,0,0,0,0,0,'Argent Valiant - In combat - Cast Shield-Breaker'),
+(@VALIANT,0,2,0,9,0,100,0,8,25,7000,7000,11,63010,0,0,0,0,0,2,0,0,0,0,0,0,0,'Argent Valiant - Distance - Cast Charge'),
+(@VALIANT,0,3,0,0,0,100,0,10000,10000,5000,10000,25,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Argent Valiant - In combat - flee'),
+(@VALIANT,0,4,5,2,0,100,1,0,5,0,0,2,35,0,0,0,0,0,1,0,0,0,0,0,0,0,'Argent Valiant - Defeated - Set faction default'),
+(@VALIANT,0,5,6,61,0,100,1,0,0,0,0,85,64885,0,0,0,0,0,7,0,0,0,0,0,0,0,'Argent Valiant - Defeated - Invoker cast spell'),
+(@VALIANT,0,6,0,61,0,100,1,0,0,0,0,80,@VALIANT*100,2,0,0,0,0,1,0,0,0,0,0,0,0,'Argent Valiant - Defeated - Run script'),
+(@VALIANT,0,7,0,11,0,100,1,0,571,210,0,46,25,0,0,0,0,0,1,0,0,0,0,0,0,0,'Argent Valiant - Just summoned - Walk 25 yards'),
+(@VALIANT*100,9,0,0,0,0,100,0,0,0,0,0,33,38595,0,0,0,0,0,7,0,0,0,0,0,0,0,'Argent Valiant - Script - Give quest credit'),
+(@VALIANT*100,9,1,0,0,0,100,0,3000,3000,0,0,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,'Argent Valiant - Script - Say text'),
+(@VALIANT*100,9,2,0,0,0,100,0,8000,8000,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Argent Valiant - Script - Despawn');
+
+DELETE FROM `creature_text` WHERE `entry`=@VALIANT;
+INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`comment`) VALUES
+(@VALIANT,0,0,"The rank of valiant is not easily won. Prepare yourself!",12,0,100,0,3,0,'Argent Valiant Aggro'),
+(@VALIANT,0,1,"You believe yourself ready for the rank of valiant? We shall see.",12,0,100,0,3,0,'Argent Valiant Aggro'),
+(@VALIANT,1,0,"Impressive skills indeed. I believe that you are more than ready for the rank of valiant.",12,0,100,0,3,0,'Argent Valiant Defeated'),
+(@VALIANT,1,1,"Well done! It seems that you are worthy of the valiant rank!",12,0,100,0,3,0,'Argent Valiant Defeated');
+
+DELETE FROM `waypoints` WHERE `entry`=@VALIANT;
+INSERT INTO `waypoints` (`entry`,`pointid`,`position_x`,`position_y`,`position_z`,`point_comment`) VALUES
+(@VALIANT,1,8575.451,952.472,547.554,'Argent Valiant'),
+(@VALIANT,2,8599.258,963.951,547.553,'Argent Valiant');
+
+-- Squire David SAI
+SET @DAVID := 33447;
+UPDATE `creature_template` SET `AIName`='SmartAI', `ScriptName`='' WHERE `entry`=@DAVID;
+
+UPDATE `gossip_menu` SET `entry`=10339 WHERE `text_id`=14407;
+DELETE FROM `gossip_menu_option` WHERE `menu_id`=10339;
+INSERT INTO `gossip_menu_option` (`menu_id`,`id`,`option_icon`,`option_text`,`option_id`,`npc_option_npcflag`,`action_menu_id`,`action_poi_id`,`box_coded`,`box_money`,`box_text`) VALUES
+(10339,0,0,'I am ready to fight!',1,1,0,0,0,0,NULL),
+(10339,1,0,'How do the Argent Crusader raiders fight?',1,1,10341,0,0,0,NULL);
+
+DELETE FROM `gossip_menu` WHERE `entry`=10341;
+INSERT INTO `gossip_menu` (`entry`,`text_id`) VALUES (10341,14410);
+DELETE FROM `npc_text` WHERE `ID`=14410;
+INSERT INTO `npc_text` (`ID`,`text0_0`,`text0_1`,`text1_0`,`text1_1`,`text2_0`,`text2_1`,`text3_0`,`text3_1`,`text4_0`,`text4_1`,`text5_0`,`text5_1`,`text6_0`,`text6_1`,`text7_0`,`text7_1`) VALUES
+(14410,'The Argent Crusade raiders believe in balance and precision, deftly mixing attack and defense.$B$BHowever, they will often open themselves to Charges by trying to gain range on you.$B$BClose that range with a Charge before they can do the same, and then Shield-Break them as you wheel around for more Thrusts.','','','','','','','','','','','','','','','');
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@DAVID AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@DAVID,0,0,0,64,0,100,0,0,0,0,0,98,10339,14407,0,0,0,0,7,0,0,0,0,0,0,0,'Squire David - Gossip Hello - Send different gossip (see conditions)'),
+(@DAVID,0,1,2,62,0,100,0,10339,0,0,0,12,@VALIANT,8,0,0,0,0,8,0,0,0,8575.451,952.472,547.554,0.38,'Squire David - Gossip select - Summon Argent Valiant'),
+(@DAVID,0,2,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,'Squire David - Gossip select - Close Gossip');
+
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=22 AND `SourceEntry`=@DAVID AND `SourceId`=0;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(22,1,@DAVID,0,0,1,0,63034,0,0,0,0,'','SAI - Squire David show different menu if player mounted'),
+(22,1,@DAVID,0,0,9,0,13679,0,0,0,0,'','SAI - Squire David show different menu if player quest taken'),
+(22,1,@DAVID,0,1,1,0,63034,0,0,0,0,'','SAI - Squire David show different menu if player mounted'),
+(22,1,@DAVID,0,1,9,0,13680,0,0,0,0,'','SAI - Squire David show different menu if player quest taken');
diff --git a/src/server/scripts/Northrend/icecrown.cpp b/src/server/scripts/Northrend/icecrown.cpp
index fb2e0f6..c323007 100644
--- a/src/server/scripts/Northrend/icecrown.cpp
+++ b/src/server/scripts/Northrend/icecrown.cpp
@@ -115,135 +115,6 @@ public:
 };
 
 /*######
-## npc_squire_david
-######*/
-
-enum eSquireDavid
-{
-    QUEST_THE_ASPIRANT_S_CHALLENGE_H                    = 13680,
-    QUEST_THE_ASPIRANT_S_CHALLENGE_A                    = 13679,
-
-    NPC_ARGENT_VALIANT                                  = 33448,
-
-    GOSSIP_TEXTID_SQUIRE                                = 14407
-};
-
-#define GOSSIP_SQUIRE_ITEM_1 "I am ready to fight!"
-#define GOSSIP_SQUIRE_ITEM_2 "How do the Argent Crusader raiders fight?"
-
-class npc_squire_david : public CreatureScript
-{
-public:
-    npc_squire_david() : CreatureScript("npc_squire_david") { }
-
-    bool OnGossipHello(Player* player, Creature* creature)
-    {
-        if (player->GetQuestStatus(QUEST_THE_ASPIRANT_S_CHALLENGE_H) == QUEST_STATUS_INCOMPLETE ||
-            player->GetQuestStatus(QUEST_THE_ASPIRANT_S_CHALLENGE_A) == QUEST_STATUS_INCOMPLETE)//We need more info about it.
-        {
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_SQUIRE_ITEM_1, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_SQUIRE_ITEM_2, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+2);
-        }
-
-        player->SEND_GOSSIP_MENU(GOSSIP_TEXTID_SQUIRE, creature->GetGUID());
-        return true;
-    }
-
-    bool OnGossipSelect(Player* player, Creature* creature, uint32 /*sender*/, uint32 action)
-    {
-        player->PlayerTalkClass->ClearMenus();
-        if (action == GOSSIP_ACTION_INFO_DEF+1)
-        {
-            player->CLOSE_GOSSIP_MENU();
-            creature->SummonCreature(NPC_ARGENT_VALIANT, 8575.451f, 952.472f, 547.554f, 0.38f);
-        }
-        return true;
-    }
-};
-
-/*######
-## npc_argent_valiant
-######*/
-
-enum eArgentValiant
-{
-    SPELL_CHARGE                = 63010,
-    SPELL_SHIELD_BREAKER        = 65147,
-
-    NPC_ARGENT_VALIANT_CREDIT   = 24108
-};
-
-class npc_argent_valiant : public CreatureScript
-{
-public:
-    npc_argent_valiant() : CreatureScript("npc_argent_valiant") { }
-
-    struct npc_argent_valiantAI : public ScriptedAI
-    {
-        npc_argent_valiantAI(Creature* creature) : ScriptedAI(creature)
-        {
-            creature->GetMotionMaster()->MovePoint(0, 8599.258f, 963.951f, 547.553f);
-            creature->setFaction(35); //wrong faction in db?
-        }
-
-        uint32 uiChargeTimer;
-        uint32 uiShieldBreakerTimer;
-
-        void Reset()
-        {
-            uiChargeTimer = 7000;
-            uiShieldBreakerTimer = 10000;
-        }
-
-        void MovementInform(uint32 uiType, uint32 /*uiId*/)
-        {
-            if (uiType != POINT_MOTION_TYPE)
-                return;
-
-            me->setFaction(14);
-        }
-
-        void DamageTaken(Unit* pDoneBy, uint32& uiDamage)
-        {
-            if (uiDamage > me->GetHealth() && pDoneBy->GetTypeId() == TYPEID_PLAYER)
-            {
-                uiDamage = 0;
-                CAST_PLR(pDoneBy)->KilledMonsterCredit(NPC_ARGENT_VALIANT_CREDIT, 0);
-                me->setFaction(35);
-                me->DespawnOrUnsummon(5000);
-                me->SetHomePosition(me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), me->GetOrientation());
-                EnterEvadeMode();
-            }
-        }
-
-        void UpdateAI(const uint32 uiDiff)
-        {
-            if (!UpdateVictim())
-                return;
-
-            if (uiChargeTimer <= uiDiff)
-            {
-                DoCastVictim(SPELL_CHARGE);
-                uiChargeTimer = 7000;
-            } else uiChargeTimer -= uiDiff;
-
-            if (uiShieldBreakerTimer <= uiDiff)
-            {
-                DoCastVictim(SPELL_SHIELD_BREAKER);
-                uiShieldBreakerTimer = 10000;
-            } else uiShieldBreakerTimer -= uiDiff;
-
-            DoMeleeAttackIfReady();
-        }
-    };
-
-    CreatureAI* GetAI(Creature* creature) const
-    {
-        return new npc_argent_valiantAI(creature);
-    }
-};
-
-/*######
 ## npc_guardian_pavilion
 ######*/
 
@@ -494,8 +365,6 @@ class npc_tournament_training_dummy : public CreatureScript
 void AddSC_icecrown()
 {
     new npc_arete;
-    new npc_squire_david;
-    new npc_argent_valiant;
     new npc_guardian_pavilion;
     new npc_vereth_the_cunning;
     new npc_tournament_training_dummy;
-- 
1.7.8.msysgit.0

